clients/src/main/java/org/apache/kafka/clients/InFlightRequests.java

```diff
@@ -35,13 +35,12 @@ public InFlightRequests(int maxInFlightRequestsPerConnection) {
     /**
      * Add the given request to the queue for the connection it was directed to
      */
-    public void add(ClientRequest request, long now) {
+    public void add(ClientRequest request) {
         Deque<ClientRequest> reqs = this.requests.get(request.request().destination());
         if (reqs == null) {
-            reqs = new ArrayDeque<ClientRequest>();
+            reqs = new ArrayDeque<>();
             this.requests.put(request.request().destination(), reqs);
         }
-        request.setSendMs(now);
```

Comment:
I could see the case for that if `InFlightRequests` had its own time source. But if it's a value passed by the caller, it reduces the cohesiveness of the class (it's no longer just a queue, it's now mutating the request) for little benefit (since all it can do is to set the value that it receives).